x/ibc-rate-limit/contracts/rate-limiter/src/packet.rs

```diff
@@ -30,35 +36,128 @@ pub struct Packet {
     pub timeout_timestamp: Option<Timestamp>,
 }
 
+// SupplyOf query message definition.
+// osmosis-std doesn't currently support the SupplyOf query, so I'm defining it localy so it can be used to obtain the channel value
+#[derive(
+    Clone,
+    PartialEq,
+    Eq,
+    ::prost::Message,
+    serde::Serialize,
+    serde::Deserialize,
+    schemars::JsonSchema,
+    CosmwasmExt,
+)]
+#[proto_message(type_url = "/cosmos.bank.v1beta1.QuerySupplyOfRequest")]
+#[proto_query(
+    path = "/cosmos.bank.v1beta1.Query/SupplyOf",
+    response_type = QuerySupplyOfResponse
+)]
+pub struct QuerySupplyOfRequest {
+    #[prost(string, tag = "1")]
+    pub denom: ::prost::alloc::string::String,
+}
```

Comment:
Wow, this is very gross.

Mentally flagging that we need to do some things to automate a lot of these derives. (Proto_message) should be able to auto-add 
```
    Clone,
    PartialEq,
    Eq,
    ::prost::Message,
    serde::Serialize,
    serde::Deserialize,
    schemars::JsonSchema,
```

cc @pyramation

Reply:
We have automation for this on osmosis-rust (which basically auto generates code like this one). That particular query was missing and it didn't generate when I ran the generation code. I passed this along to Boss so it gets added to osmosis-std, but for now I'm inlining it in my contract so we can use it right away

Reply:
SG!